lib/system.ml

```diff
@@ -53,8 +53,14 @@ module StrSet = Set.Make(StrMod)
 let dirmap = ref StrMap.empty
 
 let make_dir_table dir =
+  let entries =
+    try
+      Sys.readdir dir
+    with Sys_error _ ->
+      warn_cannot_open_dir dir;
+      Array.of_list [] in
```

Comment:
Why “Array.of_list []” rather than the empty array literal “[||]”?

Reply:
Yup, good point, @spanjel , care to amend that part?